pkg/container-collection/options.go

```diff
@@ -0,0 +1,254 @@
+		if err != nil {
+			log.Errorf("cannot start Kubernetes client: %s", err)
+			return
+		}
+		cc.k8sClient = clientset
+
+		// Future containers
+		cc.containerEnrichers = append(cc.containerEnrichers, func(containerDefinition *pb.ContainerDefinition, initial bool) {
+			// No need for this enricher if the data is already there
+			if containerDefinition.Podname != "" {
+				return
+			}
+
+			if containerDefinition.CgroupV1 == "" || containerDefinition.CgroupV2 == "" {
+				log.Errorf("Kubernetes enrichment cannot work without cgroup paths")
+				return
+			}
+
+			fieldSelector := fields.OneTermEqualSelector("spec.nodeName", nodeName).String()
+			pods, err := cc.k8sClient.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{
+				FieldSelector: fieldSelector,
+			})
+			if err != nil {
+				log.Errorf("cannot fetch pods: %s", err)
+				return
+			}
+
+			// Fill Kubernetes fields
+			namespace := ""
+			podname := ""
+			containerName := ""
+			labels := []*pb.Label{}
+			for _, pod := range pods.Items {
+				uid := string(pod.ObjectMeta.UID)
+				// check if this container is associated to this pod
+				uidWithUnderscores := strings.ReplaceAll(uid, "-", "_")
+
+				if !strings.Contains(containerDefinition.CgroupV2, uidWithUnderscores) &&
+					!strings.Contains(containerDefinition.CgroupV2, uid) &&
+					!strings.Contains(containerDefinition.CgroupV1, uidWithUnderscores) &&
+					!strings.Contains(containerDefinition.CgroupV1, uid) {
+					continue
+				}
+
+				namespace = pod.ObjectMeta.Namespace
+				podname = pod.ObjectMeta.Name
+
+				for k, v := range pod.ObjectMeta.Labels {
+					labels = append(labels, &pb.Label{Key: k, Value: v})
+				}
+				for _, container := range pod.Spec.Containers {
+					for _, mountSource := range containerDefinition.MountSources {
+						pattern := fmt.Sprintf("pods/%s/containers/%s/", uid, container.Name)
+						if strings.Contains(mountSource, pattern) {
+							containerName = container.Name
+							break
+						}
+					}
+				}
+			}
+
+			containerDefinition.Namespace = namespace
+			containerDefinition.Podname = podname
+			containerDefinition.Name = containerName
+			containerDefinition.Labels = labels
+
+			// drop pause containers
+			if containerDefinition.Podname != "" && containerName == "" {
+				containerDefinition.Id = ""
+			}
+		})
+	}
+}
+
+func WithRuncFanotify() ContainerCollectionOption {
+	return func(cc *ContainerCollection) {
+		runcNotifier, err := runcfanotify.NewRuncNotifier(func(notif runcfanotify.ContainerEvent) {
+			switch notif.Type {
+			case runcfanotify.EVENT_TYPE_ADD_CONTAINER:
+				mountSources := []string{}
+				for _, m := range notif.ContainerConfig.Mounts {
+					mountSources = append(mountSources, m.Source)
+				}
+				containerDefinition := pb.ContainerDefinition{
+					Id:           notif.ContainerID,
+					Pid:          notif.ContainerPID,
+					MountSources: mountSources,
+				}
+
+				cc.AddContainer(&containerDefinition)
```

Comment:
Here it cannot call methods from `GadgetTracerManager` because it might not have access to it...

In my next branch update, calling `GadgetTracerManager.AddContainer()` from here will not be necessary.